rdt/transformers/base.py

```diff
@@ -1,12 +1,57 @@
 """BaseTransformer module."""
 import abc
+import contextlib
 import inspect
 import warnings
+from functools import wraps
 
 import numpy as np
 import pandas as pd
 
 
+@contextlib.contextmanager
+def set_random_states(random_states, method_name, set_model_random_state):
+    """Context manager for managing the random state.
+
+    Args:
+        random_state (int or tuple):
+            The random seed or a tuple of (numpy.random.RandomState, torch.Generator).
+        set_model_random_state (function):
+            Function to set the random state on the model.
+    """
+    original_np_state = np.random.get_state()
+    random_np_state = random_states[method_name]
```

Comment:
is `random_states` a `dict` or `int or tuple` ? Looks like a `dict` to me

Reply:
it's a dict